lib/platform/azure/index.ts

```diff
@@ -620,11 +617,42 @@ export async function mergePr(
   );
 
   try {
-    await azureApiGit.updatePullRequest(
+    const response = await azureApiGit.updatePullRequest(
       objToUpdate,
       config.repoId,
       pullRequestId
     );
+
+    let retries = 0;
+    let isClosed = response.status === PullRequestStatus.Completed;
+    while (!isClosed && retries < 5) {
+      retries += 1;
+      const sleepMs = retries * 1000;
+      logger.trace(
+        `Updated PR ${pullRequestId} to closed status but change has not taken effect yet. Sleeping for ${sleepMs} before fetching PR again to check status (retry "${retries}")`
+      );
+
+      await new Promise((resolve) => setTimeout(resolve, sleepMs));
+      pr = await azureApiGit.getPullRequestById(pullRequestId, config.project);
+      isClosed = pr.status === PullRequestStatus.Completed;
+      logger.trace(
+        `PR ${pullRequestId} status is now: ${pr.status} (${
+          PullRequestStatus[pr.status]
+        })`
+      );
```

Comment:
```suggestion
      logger.trace(
        { pullRequestId, status: pr.status },
        `PR status is now: ${PullRequestStatus[pr.status]}`
      );
```


Use logger meta, maybe need prettify